pybandits/pydantic_version_compact.py

```diff
@@ -0,0 +1,242 @@
+"""
+This snippet is a compatibility layer for pydantic v1 and v2.
+"""
+
+import re
+from typing import Any, Callable, Dict, Literal, Optional, Union
+from warnings import warn
+
+from pydantic import BaseModel, Field, NonNegativeFloat, PositiveInt, ValidationError, confloat, conint, constr
+from pydantic.version import version_info as _pydantic_version
+
+
+def _get_major_pydantic_version():
```

Comment:
You can simplify the retrieval of pydantic version as follow:

```
from pydantic import version

def _get_major_pydantic_version():
    """
    Get the major version of pydantic.
    
    Returns
    -------
    major_version : str
        The major version of pydantic.
    """
    try:
        full_version = version.VERSION  
        major_version = full_version.split(".")[0]
        return major_version
    except Exception as e:
        raise ValueError(f"Error getting Pydantic version: {e}")
```

Reply:
Gr8, thx!